cmp/grid/GridModel.js

```diff
@@ -217,29 +210,26 @@ export class GridModel {
      * @param {Object} params - passed to agGrid's export functions.
      */
     localExport(filename, type, params = {}) {
-        if (!this.agApi) return;
+        const {isReady, agApi} = this.agGridModel;
+        if (!isReady) return;
         defaults(params, {fileName: filename, processCellCallback: this.formatValuesForExport});
 
         if (type === 'excel') {
-            this.agApi.exportDataAsExcel(params);
+            agApi.exportDataAsExcel(params);
         } else if (type === 'csv') {
-            this.agApi.exportDataAsCsv(params);
+            agApi.exportDataAsCsv(params);
         }
     }
 
     /** Select the first row in the grid. */
     selectFirst() {
-        const {agApi, selModel} = this;
+        const {agGridModel, selModel} = this;
+        warnIf(!agGridModel.isReady, 'Called selectFirst before the grid was ready!');
 
```

Comment:
Should we bail out here, if we have already warned?  Otherwise what's the sense of this warning?